### PR TITLE
ci: switch to github actions

### PR DESCRIPTION
**Pull request by wass3r**
## Proposed change

Originating from a breakage in docker automated build, the rabbit hole led to reimplement pipeline via Github Actions. Please review. Much is done via `make` in part to be able to replicate (most) tasks easily locally if needed. Please review! Let me know if something is off.

Side note: implemented equivalent linting that we had before. Planning to extend the things `golangci-lint` covers at a later point.

Other notable changes in this PR:
- Dockerfile
  - Changed build argument nomenclature
  - Added layer to cache dependencies
- Docker Images
  - Push to master will result in `:latest` and `:<language>` images being deployed
  - Tag will deploy versioned images and update `:latest`/`:<language>`
- Build
  - Added `-s -w` flag to build to result in smaller executables
- Github Release
  - Windows filename is without `.exe` extension at the moment
  - Added file checksum
- Makefile
  - Pretty much an overhaul

## Types of changes

What types of changes is this pull request introducing to flottbot? _Put an `x` in the boxes that apply_

- [ ] Bugfix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Checklist

You can fill this out after creating your PR. _Put an `x` in the boxes that apply_

- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I have added necessary documentation (if appropriate)

### CODE REVIEW

**Comment by wass3r** [comment]
might have to check out coverage jump.. that seems wrong

**Comment by wass3r** [comment]
so coverage was only calculating coverage based on files that had tests associated with them. went back to previous format in travis, but still getting odd results. `/core` folder results are most peculiar.

**Comment by wass3r** [comment]
switched to use [go-acc](https://github.com/ory/go-acc) and seem to be accurately reporting now though it's a tad higher than before.

File: Makefile [hunk 1]
```diff
@@ -1,36 +1,177 @@
...
+build: clean
+	@echo "Building flottbot binary to './flottbot'"
+	@GO111MODULE=on go build -a \
+		-ldflags '$(BUILD_LDFLAGS)' -o $(PWD)/flottbot ./cmd/flottbot
 
-docker:
-	docker build \
-		--build-arg "SOURCE_BRANCH=$(SOURCE_BRANCH)" \
-		--build-arg "SOURCE_COMMIT=$(SOURCE_COMMIT)" \
+.PHONY: build-cross
```
Comment (JordanSussman) [inline Makefile]: Could consider switching to [goreleaser](https://github.com/goreleaser/goreleaser-action) to simplify the building and releasing workflow.

File: docker/Dockerfile [hunk 2]
```diff
@@ -1,18 +1,25 @@
...
-ARG SOURCE_COMMIT
-WORKDIR /go/src/github.com/target/flottbot/
-RUN apk add --no-cache git
+ARG VERSION
+ARG GIT_HASH
 ENV GO111MODULE=on
-COPY / .
+
+RUN apk add --no-cache ca-certificates
+WORKDIR /go/src/github.com/target/flottbot/
```
[the remark this answers is not in the document]
Comment (wass3r) [reply docker/Dockerfile]: we do have to have a path, but we can shorten it quite a bit.. fix coming shortly

File: docker/Dockerfile.ruby [hunk 3]
```diff
@@ -1,18 +1,25 @@
...
 RUN go mod download
+
+COPY / .
+
 RUN CGO_ENABLED=0 GOOS=linux GOARCH=amd64 \
-    go build -ldflags "-X github.com/target/flottbot/version.Version=${SOURCE_BRANCH} -X github.com/target/flottbot/version.GitHash=${SOURCE_COMMIT}" \
+    go build -a -ldflags "-s -w -X github.com/target/flottbot/version.Version=${VERSION} -X github.com/target/flottbot/version.GitHash=${GIT_HASH}" \
     -o flottbot ./cmd/flottbot
 
 FROM ruby:2.6-alpine
```
[the remark this answers is not in the document]
Comment (wass3r) [reply docker/Dockerfile.ruby]: for sure!

File: docker/Dockerfile [hunk 4]
```diff
@@ -1,18 +1,25 @@
...
 RUN go mod download
+
+COPY / .
+
 RUN CGO_ENABLED=0 GOOS=linux GOARCH=amd64 \
-    go build -ldflags "-X github.com/target/flottbot/version.Version=${SOURCE_BRANCH} -X github.com/target/flottbot/version.GitHash=${SOURCE_COMMIT}" \
+    go build -a -ldflags "-s -w -X github.com/target/flottbot/version.Version=${VERSION} -X github.com/target/flottbot/version.GitHash=${GIT_HASH}" \
     -o flottbot ./cmd/flottbot
 
 FROM alpine:3.10
```
[the remark this answers is not in the document]
Comment (wass3r) [reply docker/Dockerfile]: for sure!

File: docker/Dockerfile.golang [hunk 5]
```diff
@@ -1,18 +1,26 @@
...
-ARG SOURCE_COMMIT
-WORKDIR /go/src/github.com/target/flottbot/
-RUN apk add --no-cache git
+ARG VERSION
+ARG GIT_HASH
 ENV GO111MODULE=on
-COPY / .
+
+RUN apk add --no-cache ca-certificates
+WORKDIR /go/src/github.com/target/flottbot/
```
[the remark this answers is not in the document]
Comment (wass3r) [reply docker/Dockerfile.golang]: we do have to have a path, but we can shorten it quite a bit.. fix coming shortly

File: Makefile (hunk 1 above)
Comment (wass3r) [reply Makefile]: i did consider it.. was a little hesitant to throw another config file in the root for goreleaser.. i'll take a look though as i'm not super familiar with it

Comment (wass3r) [reply Makefile]: alright, now with goreleaser :D

**Pull request merged**